fix: enable parsing of multi-option 'key:value's for config options

thorn3r opened the pull request:
<!-- Description of change -->
This fixes support for multi-option 'key:value's used for config options when only a single top-level key-value is provided, such as:
```
--api-rate-limit endpoint-create=rate-limit:2/s,rate-burst:4
```
    
Fixes: #22233
Fixes: 1280d4bc0065 ("cmd: Allow more complicated patterns in map string type.")


```release-note
Fix bug where configuring the API rate limiter options could fail when providing multiple options
```

Fixes: 070ded019adbfee49d73bc7be0c6ba3bac0ef59c not 1280d4bc0065

michi-covalent reviewed (approved):
lgtm, thanks for also including unit tests 💯 

in terms of working around this issue for existing releases, i guess you could specify the rate limiter config as JSON? something like:
```
api-rate-limit: |-
  {"endpoint-create": "rate-limit:2/s,rate-burst:4"}
```

ref: https://github.com/cilium/cilium/pull/18478

christarazi reviewed (commented):
Thanks for the PR. I didn't review the changes, but wanted to note that typically word the release-note in the PR from the perspective of the user. Since this is a bug (and a regression as well), we'll want to word it so that a user looking over the changelog can understand rather than a developer. What this usually means is that we describe what the user impact of the bug is. 

In this case, the PR title can stand as the "developer-facing impact" while the user-facing impact can be described something like:

```
Fix bug where configuring the API rate limiter options failed when providing multiple options
```

Another thing is that when we find & fix regressions, we typically link back to the commit & PR that this PR is fixing, so that we can track whether we need to backport this PR to older branches to fix the regression. This is described [here](https://docs.cilium.io/en/latest/contributing/development/contributing_guide/#submit-pr) in the docs, specifically step 6.

thorn3r committed:
fix: correct parsing of multi-option 'key:value's for config options

This fixes support for multi-option 'key:value's used for config
options when only a single top-level key-value is provided, such as
'--api-rate-limit endpoint-create=rate-limit:2/s,rate-burst:4'.

Fixes: #22233
Fixes: 070ded019adb ("cmd: Allow more complicated patterns in map string type.")

Signed-off-by: Tim Horner <timothy.horner@isovalent.com>

michi-covalent reviewed (approved):
🚢
